cmds/modules/noded/main.go

```diff
@@ -191,6 +192,42 @@ func action(cli *cli.Context) error {
 		return version, nil
 	})
 
+	bus.WithHandler("zos.system.diagnostics", func(ctx context.Context, payload []byte) (interface{}, error) {
+		type moduleStatus struct {
+			Status zbus.Status `json:"status,omitempty"`
+			Err    error       `json:"error,omitempty"`
+		}
+		results := struct {
+			SystemStatusOk bool                    `json:"system_status_ok"`
+			Modules        map[string]moduleStatus `json:"modules"`
+		}{
+			SystemStatusOk: true,
+			Modules:        make(map[string]moduleStatus),
+		}
+
+		for module := range zbusdebug.PossibleModules {
+			status, err := redis.Status(ctx, module)
+			moduleStatus := moduleStatus{
+				Status: status,
+				Err:    err,
+			}
+
+			if err != nil {
+				results.SystemStatusOk = false
+			} else {
```

Comment:
no need to check on the workers' status, I double checked with Azmy, being in Free/Busy state doesn't indicate any error